### ISSUE TITLE
re-initialization of FactoryVec seems not to work

### ISSUE THREAD

**Issue by tronta** [AaronErhardt/Relm4#84]
In my Sudoku Solver example, I wanted to reinitialize/reset of a 
https://github.com/AaronErhardt/Relm4/pull/83/files#diff-d648c225a07e9a1e3276b116cdc0265e8f74c9ebe8d453db85e8920ab5aa4f0cR78

I tried it in two ways:
* assigning the FactoryVec a new vector `self.values = AppModel::empty().values;`
* clearing it before beforehand with `self.values.clear();`

The result was in both ways that new widgets were added, but the old ones not removed.

**Comment by AaronErhardt** [comment]
Can you try again with the latest commits? The first solution won't work because I can't properly implement `Drop`for the `FactoryVec`, but the second one should now be fixed.

**Comment by tronta** [comment]
```
              self.values.clear();
              self.values = AppModel::new().values;
```
did not work. It's still adding new widget on top the others.

**Comment by AaronErhardt** [comment]
Yeah, you can't do the second line. When the factory is dropped the widgets are still alive. The only thing that works it calling `.clear()` and then re-render the factory in `view()`. As I said, I can't simply implement `Drop` so dropping the factory is almost always wrong.

**Comment by AaronErhardt** [comment]
I've added this to the 0.5 milestone. I will try to update factories so they support `Drop` which will fix this problem.
